pkg/agent/nodeportlocal/npl_agent_test.go

```diff
@@ -389,24 +390,28 @@ func TestPodDelete(t *testing.T) {
 }
 
 // TestPodAddMultiPort creates a Pod with multiple ports and verifies that the Pod's NPL annotation
-// and the local port table are updated correctly.
+// and the local port table are updated with only one port used as target port of the Service.
```

Comment:
that kind of defeats the original purpose of the test, now that we install forwarding rules for a single port...

maybe we should have a different test case to validate that *only* ports used as Service target ports are exposed with NPL

Reply:
Yes. The original test case has been modified. 
Now we are validating that only ports used as Service target ports are exposed with NPL. For annotation, this was already being done. Added couple of more checks to ensure no NPL rules are added for pod container ports.